network/src/peer/mod.rs

```diff
@@ -476,6 +461,67 @@ where
         Ok(())
     }
 
+    async fn handle_inbound_stream_message(
+        &mut self,
+        message: StreamMessage,
+    ) -> Result<(), PeerManagerError> {
+        match message {
+            StreamMessage::Header(header) => {
+                self.inbound_stream.new_stream(header)?;
```

Comment:
Silly question: are we assuming that there's only a single inbound stream active at any given time? i.e., `new_stream` below just replaces the existing in-memory stream. Is it possible that a single peer tries to stream two messages to us concurrently, e.g., state sync and mempool at the same time?

Reply:
yes, there's only one inbound stream with this implementation. it's not possible to start concurrent stream, the multiplex only supports between stream and single frame message